content/en/docs/tasks/tools/included/kubectl_installs_linux.md

```diff
@@ -0,0 +1,136 @@
+---
+title: "linux kubectl install"
+description: "Content for the Linux tab in the main page."
+toc_hide: true
+---
+
+The following methods exist for installing kubectl on Linux:
+
+{{% toc %}}
+
+## Install kubectl binary with curl on Linux
+
+1. Download the latest release with the command:
+
+   ```bash
+   curl -LO "https://dl.k8s.io/release/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl"
+   ```
+
+   {{< note >}}
+To download a specific version, replace the `$(curl -L -s https://dl.k8s.io/release/stable.txt)` portion of the command with the specific version.
+
+For example, to download version {{< param "fullversion" >}} on Linux, type:
+
+   ```bash
+   curl -LO https://dl.k8s.io/release/{{< param "fullversion" >}}/bin/linux/amd64/kubectl
+   ```
+   {{< /note >}}
+
+1. Validate the binary (optional)
+
+   Download the kubectl checksum file:
+
+   ```bash
+   curl -LO "https://dl.k8s.io/$(curl -L -s https://dl.k8s.io/release/stable.txt)/bin/linux/amd64/kubectl.sha256"
+   ```
+
+   Validate the kubectl binary against the checksum file:
+
+   ```bash
+   echo "$(<kubectl.sha256) kubectl" | sha256sum --check
+   ```
+
+   If valid, the output is:
+
+   ```bash
+   kubectl: OK
+   ```
+
+   If the check fails, `sha256` exits with nonzero status and prints output similar to:
+
+   ```bash
+   kubectl: FAILED
+   sha256sum: WARNING: 1 computed checksum did NOT match
+   ```
+
+   {{< note >}}
+   Download the same version of the binary and checksum.
+   {{< /note >}}
+
+1. Install kubectl
+
+   ```bash
+   sudo install -o root -g root -m 0755 kubectl /usr/local/bin/kubectl
+   ```
+
+   {{< note >}}
+   If you do not have root access on the target system, you can still install kubectl to the `~/.local/bin` directory:
+
+   ```bash
+   mkdir -p ~/.local/bin/kubectl
+   mv ./kubectl ~/.local/bin/kubectl
+   # and then add ~/.local/bin/kubectl to $PATH
+   ```
+
+   {{< /note >}}
+
+1. Test to ensure the version you installed is up-to-date:
+
+   ```bash
+   kubectl version --client
+   ```
+
+## Install using native package management
+
+{{< tabs name="kubectl_install" >}}
```

Comment:
I'm kinda conservative to using tabs. Tabs are good when you have more than three altertives to perfom an operation. However, tabs are not so good for presenting alternatvies that are less than three.

Another dimension is about content length. When each alternative have contents of similar length which can be contained in one screen, tabs are good. However, tabs are not suitable for the following cases:

- when the content is very short (e.g. just a dozen of lines)
- when the options have unbalanced content lengths (e.g. 10 lines versus 50 lines).
- when the content of a single option is longer than one screen -- in this case, it would be difficult to figure out where is the end of a the tabs after you scroll down the page.

Reply:
Thanks, @tengqm - just FYI this particular tabs section and the others within each OS tab aren't new / created by me - they are already in the exiting page (e.g. for this one you've highlighted: https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-using-native-package-management).

re. the content length, the included pages for the new OS tabs are relatively similar:
- Linux: 137 lines
- macOS: 125 lines
- Windows: 145 lines

Reply:
Alright, I brought this up because there are a few places where the tabs is kind of abused. For example, the following tabs are too long, it is hard to tell where the "Bash on Linux" tab ends.

https://kubernetes.io/docs/tasks/tools/install-kubectl/#kubectl-autocompletion-0

Another place where the tabs is unnecessary is:

https://kubernetes.io/docs/tasks/tools/install-kubectl/#kubectl-win-install-0

Reply:
As a writer, I tend to agree that some of the tabs aren't really necessary, but as a reader, where a tab ends doesn't really concern me as I would just carry on reading down the doc. 

Also, the second one you shared is very short and so removing it wouldn't change much, but for the autocompletion one I think a ToC list within the "Enabling shell autocompletion" section would make navigation/reading easier if the tabs were removed - otherwise it would become a very long set of text, much of which would be irrelevant for that particular reader.

Reply:
> it is hard to tell where the "Bash on Linux" tab ends.

This is a problem with the way the tabs are rendered -- I wonder if we can change tabs to have a more contained visual look. Something like a line at the bottom of them, or maybe a subtle change background colour to show what is a part of the tab content. Currently the only indicator of tab content is the horizontal one across the top of them.

Reply:
@nate-double-u +1 to the idea of a more obvious tab container - shall I open an issue for that?

Reply:
Thanks @tomkivlin, I think we should

Reply:
Done - https://github.com/kubernetes/website/issues/26737